src/test/java/tatracker/logic/commands/sort/SortModuleIntegrationTest.java

```diff
@@ -0,0 +1,176 @@
+    public void setUp() {
+        Module validModule = new ModuleBuilder().build();
+        model = new ModelManager(getTypicalTaTracker(), new UserPrefs());
+        model.addModule(validModule);
+    }
+
+    @Test
+    public void execute_groupEditedByModelAlphabetic_sortSuccessful() throws Exception {
+        Module validModule = new ModuleBuilder().withIdentifier("CS2103").build();
+        Group validGroupOne = new GroupBuilder().withIdentifier("G03").withStudent(BOB).withStudent(ALICE)
+                .withStudent(AMY).withStudent(DANIEL)
+                .withStudent(GEORGE).withStudent(FIONA).build();
+        Group validGroupTwo = new GroupBuilder().withIdentifier("G06").withStudent(BOB).withStudent(ALICE)
+                .withStudent(AMY).withStudent(DANIEL)
+                .withStudent(GEORGE).withStudent(FIONA).build();
+        validModule.addGroup(validGroupOne);
+        validModule.addGroup(validGroupTwo);
+        model.addModule(validModule);
+
+        CommandResult commandResult = new SortModuleCommand(SortType.ALPHABETIC,
+                validModule.getIdentifier()).execute(model);
+
+        assertEquals(String.format(SortModuleCommand.MESSAGE_SORT_MODULE_SUCCESS,
+                validModule.getIdentifier()), commandResult.getFeedbackToUser());
+
+
+        ObservableList<Student> students =
+                model.getModule(validModule.getIdentifier())
+                        .getGroup(validGroupOne.getIdentifier()).getStudentList();
+
+        assertEquals(ALICE, students.get(0));
+        assertEquals(AMY, students.get(1));
+        assertEquals(BOB, students.get(2));
+        assertEquals(DANIEL, students.get(3));
+        assertEquals(FIONA, students.get(4));
+        assertEquals(GEORGE, students.get(5));
+
+        students = model.getModule(validModule.getIdentifier())
+                        .getGroup(validGroupTwo.getIdentifier()).getStudentList();
+
+        assertEquals(ALICE, students.get(0));
+        assertEquals(AMY, students.get(1));
+        assertEquals(BOB, students.get(2));
+        assertEquals(DANIEL, students.get(3));
+        assertEquals(FIONA, students.get(4));
+        assertEquals(GEORGE, students.get(5));
+    }
+
+    @Test
+    public void execute_groupEditedByModelRatingAsc_sortSuccessful() throws Exception {
+        Module validModule = new ModuleBuilder().withIdentifier("CS2103").build();
+        Group validGroupOne = new GroupBuilder().withIdentifier("G03").withStudent(BOB).withStudent(ALICE)
+                .withStudent(AMY).withStudent(DANIEL)
+                .withStudent(GEORGE).withStudent(FIONA).build();
+        Group validGroupTwo = new GroupBuilder().withIdentifier("G06").withStudent(BOB).withStudent(ALICE)
+                .withStudent(AMY).withStudent(DANIEL)
+                .withStudent(GEORGE).withStudent(FIONA).build();
+        validModule.addGroup(validGroupOne);
+        validModule.addGroup(validGroupTwo);
+        model.addModule(validModule);
```

Comment:
Next time can save this outside in the Before each, then ur test cases will be smaller!